### PR TITLE
fix(overlay)!: fix setting model id in overlay (Issue #2297)

### PR DESCRIPTION
**Pull request by denys-kolomiitsev**
**Description:**

Fixed creation of the new conversation in overlay if modelId set in the overlay settings

Issues:

- Issue #2297 

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Review by Alexander-Kezik — DISMISSED** [review]

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2604.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2604.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2604.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2604.nightly-test.deltixhub.io
>E2E tests status: failed

File: apps/chat/src/store/models/models.epics.ts [hunk 1]
```diff
@@ -375,10 +375,18 @@ const updateRecentModelsEpic: AppEpic = (action$, state$) =>
...
   action$.pipe(
     filter(ModelsActions.getModelsSuccess.match),
     switchMap(({ payload }) => {
+      const isOverlay = SettingsSelectors.selectIsOverlay(state$.value);
+      const isOverlayDefaultModelId =
+        SettingsSelectors.selectIsOverlayDefaultModelId(state$.value);
+
+      if (isOverlay && isOverlayDefaultModelId) {
+        return of(ModelsActions.getInstalledModelIds());
+      }
```
Comment (IlyaBondar) [inline apps/chat/src/store/models/models.epics.ts]: why do we return installed model ids in this case?

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2604.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/store/models/models.epics.ts (hunk 1 above)
Comment (denys-kolomiitsev) [reply apps/chat/src/store/models/models.epics.ts]: Changed to return EMPTY.
Thank you for the suggestion.

File: apps/chat/README.md [hunk 2]
```diff
@@ -115,7 +115,7 @@ AI DIAL Chat uses environment variables for configuration. All environment varia
...
+| `DEFAULT_MODEL`                           |    No    | A model that will be used for the new conversation if the recently-used models or recently-used models from local storage are not available. If none of the above is available - the first model from the AI DIAL Core config will be used as a pre-selected conversation model.                                              | Any string                                                                                       | First available model from [AI DIAL Core](https://github.com/epam/ai-dial-core?tab=readme-ov-file#dynamic-settings) config listing |
```
Comment (IlyaBondar) [inline apps/chat/README.md:118]: `from local storage`?)

File: apps/chat/src/store/models/models.epics.ts [hunk 3]
```diff
@@ -375,10 +375,18 @@ const updateRecentModelsEpic: AppEpic = (action$, state$) =>
...
     filter(ModelsActions.getModelsSuccess.match),
     switchMap(({ payload }) => {
+      const isOverlay = SettingsSelectors.selectIsOverlay(state$.value);
+      const overlayDefaultModelId =
+        SettingsSelectors.selectOverlayDefaultModelId(state$.value);
+
+      if (isOverlay && overlayDefaultModelId) {
+        return EMPTY;
+      }
+
```
Comment (IlyaBondar) [inline apps/chat/src/store/models/models.epics.ts:389]: for what this block?

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2604.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2604.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2604.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2604.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2604.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/store/models/models.epics.ts [hunk 4]
```diff
@@ -375,13 +375,16 @@ const updateRecentModelsEpic: AppEpic = (action$, state$) =>
...
     filter(ModelsActions.getModelsSuccess.match),
     switchMap(({ payload }) => {
-      const defaultModelId = payload.models.find(
-        (model) => model.isDefault,
-      )?.id;
+      const overlayDefaultModelId =
+        SettingsSelectors.selectOverlayDefaultModelId(state$.value);
+
+      const defaultModelId = overlayDefaultModelId
+        ? undefined
```
Comment (IlyaBondar) [inline apps/chat/src/store/models/models.epics.ts:386]: why do you return `undefined` here?

**Review by IlyaBondar — APPROVED** [review]

File: apps/chat/src/store/models/models.epics.ts (hunk 4 above)
Comment (denys-kolomiitsev) [reply apps/chat/src/store/models/models.epics.ts:386]: We need it to skip to skip assignment of default model and array.find() run.

**Review by irinakartun — APPROVED** [review]

**Pull request merged**